dot/types/grandpa.go

```diff
@@ -196,3 +196,61 @@ type GrandpaVote struct {
 func (v GrandpaVote) String() string {
 	return fmt.Sprintf("hash=%s number=%d", v.Hash, v.Number)
 }
+
+// GrandpaEquivocation is used to create a proof of equivocation
+type GrandpaEquivocation struct {
+	RoundNumber     uint64
+	ID              [32]byte
+	FirstVote       GrandpaVote
+	FirstSignature  [64]byte
+	SecondVote      GrandpaVote
+	SecondSignature [64]byte
+}
+
+// GrandpaEquivocationVote is a custom vdt type for a grandpa equivocation
+type GrandpaEquivocationVote scale.VaryingDataType
```

Comment:
this makes it sound like it represents vote, which it doesn't.
rename to something like GrandpaEquivocationEnum or GrandpaEquivocationWrapper